src/lsptoolshost/roslynLanguageServer.ts

```diff
@@ -472,40 +442,75 @@ export class RoslynLanguageServer {
 
         // Razor will call into us (via command) for generated file didChange/didClose notifications. We'll then forward these
         // notifications along to Roslyn. didOpen notifications are handled separately via the vscode.openTextDocument method.
-        vscode.commands.registerCommand(RoslynLanguageServer.roslynDidChangeCommand, (notification: DidChangeTextDocumentParams) => {
+        client.addDisposable(vscode.commands.registerCommand(RoslynLanguageServer.roslynDidChangeCommand, (notification: DidChangeTextDocumentParams) => {
```

Comment:
before, on restarts we were trying to register the command again.  Now we register the command as a disposable that will get disposed of when the server shuts down, so we can register it again on the new instance.

Reply:
Is there a reason we're not just registering these during extension activation (assuming we're not in Omnisharp mode) and instead having some other indirection around ensuring they're hitting the current client?

Reply:
@allisonchou @davidwengier do you know?  Do these need to be registered after the server is created, or can I move them?

Reply:
I'll move just the command registration earlier.  I don't think it can hurt if the commands are being registered earlier (if they were being called they would throw with missing command before, now they'll throw with server not started instead)

Reply:
No specific knowledge from me.

Reply:
Yep, I think that change should be fine!